go/integTest/readfrom_strategy_test.go

```diff
@@ -0,0 +1,410 @@
+		WithAddress(&suite.clusterHosts[0]).
+		WithUseTLS(suite.tls).
+		WithRequestTimeout(2000).
+		WithReadFrom(api.AzAffinity).
+		WithClientAZ(az))
+
+	azGetResult, err := clientForTestingAz.CustomCommandWithRoute(
+		[]string{"CONFIG", "GET", "availability-zone"}, config.AllNodes)
+	assert.NoError(suite.T(), err)
+	for _, value := range azGetResult.MultiValue() {
+		if valueMap, ok := value.(map[string]interface{}); ok {
+			if azValue, ok := valueMap["availability-zone"].(string); ok {
+				assert.Equal(suite.T(), az, azValue)
+			}
+		}
+	}
+
+	// Execute GET commands
+	for i := 0; i < nGetCalls; i++ {
+		_, err := clientForTestingAz.Get("foo")
+		assert.NoError(suite.T(), err)
+	}
+
+	infoResult, err := clientForTestingAz.InfoWithOptions(
+		options.ClusterInfoOptions{
+			InfoOptions: &options.InfoOptions{Sections: []options.Section{options.All}},
+			RouteOption: &options.RouteOption{Route: config.AllNodes},
+		},
+	)
+	assert.NoError(suite.T(), err)
+
+	// Check that all replicas have the same number of GET calls
+	matchingEntriesCount := 0
+	for _, value := range infoResult.MultiValue() {
+		if strings.Contains(value, getCmdStat) && strings.Contains(value, az) {
+			matchingEntriesCount++
+		}
+	}
+	assert.Equal(suite.T(), nReplicas, matchingEntriesCount)
+
+	clientForTestingAz.Close()
+}
+
+func (suite *GlideTestSuite) TestAzAffinityNonExistingAz() {
+	suite.SkipIfServerVersionLowerThanBy("8.0.0")
+
+	const nGetCalls = 3
+	const nReplicaCalls = 1
+	getCmdStat := fmt.Sprintf("cmdstat_get:calls=%d", nReplicaCalls)
+
+	clientForTestingAz := suite.clusterClient(api.NewGlideClusterClientConfiguration().
+		WithAddress(&suite.clusterHosts[0]).
+		WithUseTLS(suite.tls).
+		WithRequestTimeout(2000).
+		WithReadFrom(api.AzAffinity).
+		WithClientAZ("non-existing-az"))
+
+	// Reset stats
+	resetStatResult, err := clientForTestingAz.CustomCommandWithRoute(
+		[]string{"CONFIG", "RESETSTAT"}, config.AllNodes)
+	assert.NoError(suite.T(), err)
+	assert.Equal(suite.T(), "OK", resetStatResult.SingleValue())
+
+	// Execute GET commands
+	for i := 0; i < nGetCalls; i++ {
+		_, err := clientForTestingAz.Get("foo")
+		assert.NoError(suite.T(), err)
+	}
+
+	infoResult, err := clientForTestingAz.InfoWithOptions(
+		options.ClusterInfoOptions{
+			InfoOptions: &options.InfoOptions{Sections: []options.Section{options.Commandstats}},
+			RouteOption: &options.RouteOption{Route: config.AllNodes},
+		},
+	)
+	assert.NoError(suite.T(), err)
+
+	// We expect the calls to be distributed evenly among the replicas
+	matchingEntriesCount := 0
+	for _, value := range infoResult.MultiValue() {
+		if strings.Contains(value, getCmdStat) {
+			matchingEntriesCount++
+		}
+	}
+	assert.Equal(suite.T(), 3, matchingEntriesCount)
+
+	clientForTestingAz.Close()
+}
+
+func (suite *GlideTestSuite) TestConnectionTimeoutClusterMode() {
+	suite.testConnectionTimeout(true)
+}
+
+func (suite *GlideTestSuite) TestConnectionTimeoutStandaloneMode() {
+	suite.testConnectionTimeout(false)
+}
```

Comment:
You can parametrize it, like we do in IT suite, see `runWithClients` in `glide_test_suite_test.go`